typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -828,7 +884,7 @@
             (match (resolve ty)
               [(Function: _) #t]
               [(Union: _ elems) (andmap loop elems)]
-              [(Intersection: elems) (ormap loop elems)]
+              [(Intersection: elems _) (ormap loop elems)]
```

Comment:
same comment as above -- it's sound to ignore the refinement prop and we don't gain anything (as far as I can see) from including it